# In this file using following columns build a model to predict if person would survive or not,

Pclass

Sex

Age

Fare


In [1]:
#import libaroes
import pandas as pd
df = pd.read_csv('titanic.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [3]:
#create on dataframe of necessary features
inputs = df[['Pclass','Sex','Age','Fare']]

In [4]:
inputs

Pclass     Sex   Age     Fare
0         3    male  22.0   7.2500
1         1  female  38.0  71.2833
2         3  female  26.0   7.9250
3         1  female  35.0  53.1000
4         3    male  35.0   8.0500
..      ...     ...   ...      ...
886       2    male  27.0  13.0000
887       1  female  19.0  30.0000
888       3  female   NaN  23.4500
889       1    male  26.0  30.0000
890       3    male  32.0   7.7500

[891 rows x 4 columns]

In [5]:
target = df['Survived']
target

0      0
1      1
2      1
3      1
4      0
      ..
886    0
887    1
888    0
889    1
890    0
Name: Survived, Length: 891, dtype: int64

In [6]:
inputs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Pclass  891 non-null    int64  
 1   Sex     891 non-null    object 
 2   Age     714 non-null    float64
 3   Fare    891 non-null    float64
dtypes: float64(2), int64(1), object(1)
memory usage: 28.0+ KB


In [12]:
#Encode the sex attribute as m =1 and F =0
from sklearn.preprocessing import LabelEncoder
Sex_n = LabelEncoder()
inputs['Sex_'] = Sex_n.fit_transform(inputs['Sex'])

<ipython-input-12-a2e1762818cb>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  inputs['Sex_'] = Sex_n.fit_transform(inputs['Sex'])


In [13]:
inputs

Pclass     Sex   Age     Fare  Sex_
0         3    male  22.0   7.2500     1
1         1  female  38.0  71.2833     0
2         3  female  26.0   7.9250     0
3         1  female  35.0  53.1000     0
4         3    male  35.0   8.0500     1
..      ...     ...   ...      ...   ...
886       2    male  27.0  13.0000     1
887       1  female  19.0  30.0000     0
888       3  female   NaN  23.4500     0
889       1    male  26.0  30.0000     1
890       3    male  32.0   7.7500     1

[891 rows x 5 columns]

In [19]:
#drop unencoded feautres
inputs_n = inputs.drop(['Sex'], axis ='columns')

In [20]:
inputs_n

Pclass   Age     Fare  Sex_
0         3  22.0   7.2500     1
1         1  38.0  71.2833     0
2         3  26.0   7.9250     0
3         1  35.0  53.1000     0
4         3  35.0   8.0500     1
..      ...   ...      ...   ...
886       2  27.0  13.0000     1
887       1  19.0  30.0000     0
888       3   NaN  23.4500     0
889       1  26.0  30.0000     1
890       3  32.0   7.7500     1

[891 rows x 4 columns]

In [24]:
#ti fill nan vallues of age with age mean
inputs_n.Age = inputs_n.Age.fillna(inputs_n.Age.mean())

In [25]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test= train_test_split(inputs_n,target,test_size=0.2)

In [26]:
from sklearn import tree
model = tree.DecisionTreeClassifier()

In [27]:
model.fit(x_train, y_train)

DecisionTreeClassifier()

In [28]:
model.score(x_test,y_test)

0.8100558659217877

# Conclusion

The above model of decision tree got 81 % accuracy